app/src/main/java/org/dhis2/usescases/teiDashboard/TeiDashboardMobileActivity.java

```diff
@@ -157,57 +181,117 @@ protected void onRestoreInstanceState(Bundle savedInstanceState) {
         outState.putString(Constants.PROGRAM_UID, programUid);
+        outState.putString(Constants.ENROLLMENT_UID, enrollmentUid);
         super.onSaveInstanceState(outState);
     }
 
     private void setViewpagerAdapter() {
 
-        for (Fragment fragment : getSupportFragmentManager().getFragments())
-            getSupportFragmentManager().beginTransaction().remove(fragment).commitNow();
-
-        binding.teiPager.setAdapter(null);
-        binding.teiPager.invalidate();
-
         if (OrientationUtilsKt.isPortrait()) {
-            adapter = new DashboardPagerAdapter(this, getSupportFragmentManager(), programUid, teiUid);
+            binding.teiPager.setAdapter(null);
+            adapter = new DashboardPagerAdapter(this, programUid, teiUid, enrollmentUid);
             currentAdapter = adapter;
             binding.teiPager.setAdapter(adapter);
-            binding.tabLayout.setVisibility(View.VISIBLE);
+            binding.teiPager.registerOnPageChangeCallback(
+                    new ViewPager2.OnPageChangeCallback() {
+                        @Override
+                        public void onPageSelected(int position) {
+                            if (position != 0 || programUid == null) {
+                                binding.filterCounter.setVisibility(View.GONE);
+                                binding.searchFilterGeneral.setVisibility(View.GONE);
+                            } else {
+                                binding.filterCounter.setVisibility(View.VISIBLE);
+                                binding.searchFilterGeneral.setVisibility(View.VISIBLE);
+                            }
+                        }
+                    }
+            );
+            //binding.tabLayout.setVisibility(programUid != null ? View.VISIBLE : View.GONE);
             if (fromRelationship)
                 binding.teiPager.setCurrentItem(2, false);
+
+            tabLayoutMediator(binding.teiPager);
         } else {
-            tabletAdapter = new DashboardPagerTabletAdapter(this, getSupportFragmentManager(), programUid, teiUid);
+            tabletAdapter = new DashboardPagerTabletAdapter(this, programUid, teiUid, enrollmentUid);
             currentAdapter = tabletAdapter;
-            binding.teiPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
-                @Override
-                public void onPageScrolled(int i, float v, int i1) {
-                    // nothing
-                }
-
-                @Override
-                public void onPageSelected(int i) {
-                    binding.sectionTitle.setText(tabletAdapter.getPageTitle(i));
-                }
+            binding.teiTablePager.registerOnPageChangeCallback(
+                    new ViewPager2.OnPageChangeCallback() {
+                        @Override
+                        public void onPageSelected(int position) {
+                            switch (position) {
+                                case 1:
```

Comment:
can we use constant for these numbers ?? for easy identification?

Reply:
Done